hail/src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -246,9 +246,9 @@ object PruneDeadFields {
               }
               TTuple(subFields)
             case ta: TArray =>
-              TArray(unifySeq(ta.elementType, children.map(_.asInstanceOf[TArray].elementType)))
+              TArray(unifySeq(ta.elementType, children.map(TIterable.elementType)))
```

Comment:
Added `TIterable.elementType` because I got bored of writing `.asInstanceOf[TIterable].elementType`. This generated some noise but has made things slighly more concise. Sorry.